opensearch_dashboards.json

```diff
@@ -10,7 +10,8 @@
     "savedObjectsManagement"
   ],
   "optionalPlugins": [
```

Comment:
TODO: building the plugin fails when adding dataSourceManagement as a "required bundle", even though it is suggested to do so by the core team. I opened an issue in core here: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/6018 to get more clarity, but removed it for now from this PR. Will follow up after core responds to the issue.

Reply:
Reverted it back since without it it a different error is thrown. Will be ok to merge this with build/install workflow failing but everything else passing, until core puts in a fix or suggests a workaround.

Reply:
Added it as an optional plugin as per suggestion from @BionIT

Reply:
Optional is not working. Leaving it as requiredBundle, even though this is failing yarn build/install workflow for now.